### PR TITLE
[mission-control] Add support for livenessProbe and livenessProbe for all microservices

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
[mission-control] Add support for livenessProbe and livenessProbe for all microservices

### CODE REVIEW

File: stable/mission-control/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # JFrog Mission-Control Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [4.3.2] - Aug 25, 2020
+* Add support for livenessProbe and livenessProbe for all Microservices.
```
Comment (danielezer) [inline stable/mission-control/CHANGELOG.md:5]: @chukka livenessProbe and livenessProbe ?

File: stable/mission-control/README.md [hunk 2]
```diff
@@ -461,8 +461,8 @@ The following table lists the configurable parameters of the mission-control cha
 | `insightServer.name`                         | Insight Server name                             | `insight-server`                      |
 | `insightServer.image.repository`             | Container image                                 | `docker.bintray.io/jfrog/insight-server`|
 | `insightServer.image.version`                | Container image tag                             | `.Chart.AppVersion`                   |
-| `insightServer.externalHttpPort`             | Insight Server service external port            | `8082`                                |
-| `insightServer.internalHttpPort`             | Insight Server service internal port            | `8082`                                |
+| `insightServer.externalPort`                 | Insight Server service external port            | `8082`                                |
```
Comment (danielezer) [inline stable/mission-control/README.md:464]: This is breaking. You need to say that in the CHANGELOG and bump the version to more than a patch

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/mission-control/CHANGELOG.md (hunk 1 above)
Comment (chukka) [reply stable/mission-control/CHANGELOG.md]: Done

File: stable/mission-control/README.md (hunk 2 above)
Comment (chukka) [reply stable/mission-control/README.md]: Done

File: stable/mission-control/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # JFrog Mission-Control Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [4.4.0] - Aug 25, 2020
+* **IMPORTANT**
```
Comment (danielezer) [inline stable/mission-control/CHANGELOG.md:5]: Say it is breaking please @chukka

File: stable/mission-control/README.md [hunk 4]
```diff
@@ -461,8 +461,7 @@ The following table lists the configurable parameters of the mission-control cha
 | `insightServer.name`                         | Insight Server name                             | `insight-server`                      |
 | `insightServer.image.repository`             | Container image                                 | `docker.bintray.io/jfrog/insight-server`|
 | `insightServer.image.version`                | Container image tag                             | `.Chart.AppVersion`                   |
-| `insightServer.externalHttpPort`             | Insight Server service external port            | `8082`                                |
```
Comment (danielezer) [inline stable/mission-control/README.md:464]: Why did you remove the external port?

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/mission-control/README.md (hunk 4 above)
Comment (chukka) [reply stable/mission-control/README.md:464]: we don't need this - Everything is via router .

Comment (danielezer) [reply stable/mission-control/README.md:464]: 👍

File: stable/mission-control/CHANGELOG.md (hunk 3 above)
Comment (danielezer) [reply stable/mission-control/CHANGELOG.md:5]: @chukka just this, please. Indicate that you are changing the `internalHttpPort` to `internalPort` and that this is breaking

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/mission-control/CHANGELOG.md (hunk 3 above)
Comment (chukka) [reply stable/mission-control/CHANGELOG.md]: Done

File: stable/mission-control/README.md (hunk 2 above)
Comment (chukka) [reply stable/mission-control/README.md]: Done

**Review by danielezer — APPROVED** [review]

File: stable/mission-control/values.yaml [hunk 5]
```diff
@@ -307,6 +307,30 @@ elasticsearch:
...
+      failureThreshold: 3
+      successThreshold: 1
+
+  readinessProbe:
+    enabled: true
+    config: |
+      httpGet:
+        path: /_cat/health
+        port: {{ .Values.elasticsearch.httpPort }}
+      initialDelaySeconds: 60
```
Comment (danielezer) [inline stable/mission-control/values.yaml:328]: It doesn't make sense that the liveness probe has a shorter `initialDelaySeconds` than the readiness probe. They should either be the same or it should be the other way around

File: stable/mission-control/values.yaml [hunk 6]
```diff
@@ -488,6 +512,32 @@ missionControl:
...
   externalPort: 80
 
+  livenessProbe:
+    enabled: true
+    config: |
+      exec:
+        command:
+        - curl 
+        - http://127.0.0.1:{{ .Values.missionControl.internalPort }}/api/v1/system/ping
+      initialDelaySeconds: 90
```
Comment (danielezer) [inline stable/mission-control/values.yaml:522]: Same here

File: stable/mission-control/values.yaml [hunk 7]
```diff
@@ -488,6 +512,32 @@ missionControl:
   internalPort: 8080
   externalPort: 80
 
+  livenessProbe:
+    enabled: true
+    config: |
+      exec:
+        command:
+        - curl 
+        - http://127.0.0.1:{{ .Values.missionControl.internalPort }}/api/v1/system/ping
```
Comment (danielezer) [inline stable/mission-control/values.yaml:521]: Why use curl? There's no guarantee that you would always have curl on the docker image. There's a tendency in JFrog to remove curl from the images every now and then ;) 
Why not use httpGet?

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/mission-control/values.yaml (hunk 7 above)
Comment (rahulsadanandan) [reply stable/mission-control/values.yaml:521]: httpGet was showing connection refused errors , since the microservices only accept connection from localhost. That is why we had to use curl

Comment (eldada) [reply stable/mission-control/values.yaml:521]: Strange... the probe is supposed to run in the context of the localhost. This might be an application bug.

Comment (danielezer) [reply stable/mission-control/values.yaml:521]: This looks pretty weird from a customer's perspective...

File: stable/mission-control/CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,12 @@
 # JFrog Mission-Control Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [5.1.0] - Oct 5, 2020
+* **Breaking**
+* Changed `insightServer.internalHttpPort` to `insightServer.internalPort`
+* Add support for livenessProbe and readinessProbe for all microservices
+* Updated UPGRADE_NOTES.md
```
Comment (chukka) [inline stable/mission-control/CHANGELOG.md:8]: @rahulsadanandan  Can we add what we changed in UPGRADE_NOTES

**Review by danielezer — APPROVED** [review]

File: stable/mission-control/values.yaml (hunk 5 above)
Comment (rahulsadanandan) [reply stable/mission-control/values.yaml:343]: `initialDelaySeconds` for liveness probe is 90 seconds 
and for readiness is 60 seconds .

File: stable/mission-control/CHANGELOG.md (hunk 8 above)
Comment (rahulsadanandan) [reply stable/mission-control/CHANGELOG.md]: Corrected.

File: stable/mission-control/values.yaml (hunk 7 above)
Comment (rahulsadanandan) [reply stable/mission-control/values.yaml:536]: As discussed, I have created a separate ticket to see if it's possible to use httpGet

**Review by rahulsadanandan — APPROVED** [review]

**Review by eldada — APPROVED** [review]
